docs/proposals/fp2.rst

```diff
@@ -0,0 +1,145 @@
+those rules to be defined within the codebase being linted, without needing to
+first install those lint rules into the environment before running Fixit.
+
+These are defined as "local rules", and will require dedicated behavior from
+Fixit to discover and import these rules at runtime, as well as special
+configuration syntax for enabling these local rules in a project.
+
+
+Configuration
+-------------
+
+When configuring the set of enabled or disabled rules, any local rules must
+be marked with a single leading period, and referred to using their path
+relative to the configuration file's directory.
+
+For example, a configuration file at ``project/fixit.toml`` could include
+rules defined in ``project/some/local/rules.py`` with the following:
+
+.. code-block:: toml
+
+    [tool.fixit]
+    enable = [".some.local.rules"]
+
+
+References to local rules should be accepted in either ``enable`` or ``disable``
+options, including overrides, to provide the same selection criteria available
+to global rules:
+
+.. code-block:: toml
+
+    [tool.fixit]
+    enable = [".rules"]
+
+    [[tool.fixit.overrides]]
+    path = "project1"
+    disable = [".rules.PickyRule"]
+    enable = [".project1.rules"]
+
+
+Implementation
+--------------
+
+When gathering enabled/disabled rules from configuration, Fixit currently
+just records the module name (fqdn) and does set operations when merging
+configs/overrides. In order to differentiate and manage local rules correctly,
+without affecting the behavior of merging/overrides, one possible option is to
+replace the simple string with a tuple of path and module.
+
+For example, where a global rule would be tracked as just ``"fixit.rules"``,
+local rules from ``foo/bar/fixit.toml`` could be tracked as
+``(Path('foo/bar'), ".local.rules"))``, allowing set operations while still
+tracking their origin, and preventing collisions from different directories.
+
+In this implementation, the path object should match the parent directory
+containing the configuration file that enables or disables these local rules.
+Options in both a ``pyproject.toml`` and ``fixit.toml`` from the same exact
+directory would use the same path object, as would any local rules referenced
+by overrides in those files.
+
+---
+
+When discovering and loading rules, the system should attempt to make sure
+that it is loading rules from the local path, rather than accidentally loading
+rules from the outside environment. This can be done either by a temporarily
+restricted path when importing, or with a custom import loader.
+
+The former may be a simpler starting point, and can be handled with a temporary
+override of ``sys.path`` to include the local directory — and nothing else? —
+and then importing the module as normal, with the leading period removed.
+
+This could look something like:
+
+.. code-block:: python
+
+    with temporary_sys_path(parent_path):
+        name = name.lstrip(".")
```

Comment:
What should two leading dots do?

Reply:
The "limitations" section mentions that we want to specifically disallow things like that. I expect that we'll have some validation when reading configs that calls that out as not allowed.